src/indextypes.jl

```diff
@@ -35,10 +35,21 @@ end
 # https://github.com/JuliaLang/julia/issues/10208
 Base.hash(v::VariableIndex, h::UInt) = hash(v.value, h)
 
+"""
+    SubmittedIndex{SubmitType}
+
+A type-safe wrapper for `Int64` for use in referencing elements submitted for
+submittable of type `SubmitType`.
+"""
+struct SubmittedIndex{SubmitType}
+    attr::SubmitType
```

Comment:
Is this necessary? What is it needed for?

Reply:
Indices of different submittable should be different to avoid mixing indices of different ones. The hole point of these indices types is to ensure that they are not used in incorrect places and adding the attribute type might help in this respect